lib/spack/spack/test/cmd/test.py

```diff
@@ -248,15 +248,21 @@ def test_hash_change(mock_test_stage, mock_packages, mock_archive, mock_fetch,
     outfile = os.path.join(testdir, 'test_suite.lock')
     with open(outfile, 'r') as f:
         output = f.read()
+        val_replace = '"hash": "{0}"'.format(
+            spack.store.db.query('printing-package')[0].dag_hash())
         changed_hash = output.replace(
-            spack.store.db.query('printing-package')[0].full_hash(),
-            'fakehash492ucwhwvzhxfbmcc45x49ha')
+            val_replace,
+            '"hash": "fakehash492ucwhwvzhxfbmcc45x49ha"')
     with open(outfile, 'w') as f:
         f.write(changed_hash)
 
     # The find command should show the contents
     find_output = spack_test('find')
     assert 'printpkg' in find_output
     # The results should be obtainable
-    results_output = spack_test('results')
+    results_output = spack_test('results', output=str)
+    print('spack test results')
+    print(results_output)
     assert 'PASSED' in results_output
```

Comment:
@nhanford Since you added this test, I thought I'd get your thoughts on what it means now (with this PR) if the `dag_hash` is computed as the `full_hash` once was, and the `full_hash` is no longer used.

The test suite abstraction seems to read itself in from files in the stage along with the test results.  Since the `dag_hash` is used in the package id that's written out to the test results, if we change the `dag_hash` in the test suite lockfile, the tested specs end up with a different package id than the results file has.

My inclination is just to delete this test, but let me know what you think.

Reply:
I suspect I could update this test to change the hash in the test results file (so it matches the hash change made in the test suite lockfile), and that might get the test to pass, but I'm not sure what benefit there is to that.  Without a described purpose (the current docstring is identical to that of `test_test_output`, another test in this file), I'm not sure what was the original intent of the test.

@nhanford If you recall any details about why you added this test, it could help me figure out whether I should fix/update it, or just remove it.  Thanks!

Reply:
When #22845 broke hashes, issue #25898 was raised regarding being unable to read old test results.
You are correct that the onlly goal of this unit test is to make sure that old test results can still be read after a hash change. I think it's fine to update the hashes so that they are coherent, as long as the old results can still be read.

Reply:
Since:

1.test result data includes a serialized spec.json (saved in the `test_suite.lock` file), and 
2. this PR should not not change the hashes of installed specs (or any serialized specs at all, for that matter), and finally
3. there is no longer a `full_hash()` method on specs that can be different from the `dag_hash`,

I think this test should instead just make sure we can always read in an old test suite and results.  So I used the current spack `develop` to run a test suite and saved the generated data in the repo.  Now this test (renamed to `test_read_old_results`) just copies that saved data into the mock stage directory and then runs `spack test find` and `spack test results` to make sure we can still read it.

Reply:
Sounds good; thanks Scott!